instrumentation/spring/spring-integration-4.1/library/src/main/java/io/opentelemetry/instrumentation/spring/integration/TracingChannelInterceptor.java

```diff
@@ -22,12 +23,16 @@
 import org.springframework.util.LinkedMultiValueMap;
 
 final class TracingChannelInterceptor implements ExecutorChannelInterceptor {
-  private static final String CONTEXT_AND_SCOPE_KEY = ContextAndScope.class.getName();
-  private static final String SCOPE_KEY = TracingChannelInterceptor.class.getName() + ".scope";
 
   private final ContextPropagators propagators;
   private final Instrumenter<MessageWithChannel, Void> instrumenter;
 
+  // TODO (trask): optimize for javaagent by using field-backed context store
+  private final Cache<MessageChannel, ContextAndScope> sendContextAndScopeHolder =
```

Comment:
Isn't `MessageChannel` a shared object that is used to send many possibly concurrent messages?

Reply:
oh no 🙁

Reply:
If the underscore idea doesn't work out, here's another take on this problem that I considered when developing this instrumentation: https://github.com/open-telemetry/opentelemetry-java-instrumentation/pull/3120/commits/7c8d614f4c9734580ab3b54fcd916383ffeffddd